api/v1alpha1/common_types.go

```diff
@@ -47,6 +47,24 @@ type ControlPlaneVirtualIPSpec struct {
 	// +kubebuilder:validation:Enum=KubeVIP
 	// +kubebuilder:default=KubeVIP
 	Provider string `json:"provider,omitempty"`
+
+	// Configuration for the chosen control-plane virtual IP provider.
+	// +kubebuilder:validation:Optional
+	Configuration *ControlPlaneVirtualIPConfiguration `json:"configuration,omitempty"`
```

Comment:
can we directly use `Address` and `Port` here?  what is motivation behind adding `Configuration` wrapper?

Reply:
Mainly for consistency with existing APIs https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/8e2b30770c64d6976a7990274988d53f9626f169/api/v1alpha1/addon_types.go#L260 and https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/8e2b30770c64d6976a7990274988d53f9626f169/api/v1alpha1/addon_types.go#L196.

Reply:
I don't think this is a necessary level of indirection in this case. The examples given are there to support generic configuration for LB and StorageClasses, but in this case this case it's a provider specific configuration for Nutanix clusters.